benchmark/cublas/blas1/rotmg.cpp

```diff
@@ -0,0 +1,168 @@
+  }
+  return;
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, cublasHandle_t* cuda_handle_ptr,
+         bool* success) {
+  // Create data
+  constexpr size_t param_size = 5;
+  std::vector<scalar_t> param = std::vector<scalar_t>(param_size);
+  scalar_t d1 = blas_benchmark::utils::random_data<scalar_t>(1)[0];
+  scalar_t d2 = blas_benchmark::utils::random_data<scalar_t>(1)[0];
+  scalar_t x1 = blas_benchmark::utils::random_data<scalar_t>(1)[0];
+  scalar_t y1 = blas_benchmark::utils::random_data<scalar_t>(1)[0];
+
+  cublasHandle_t& cuda_handle = *cuda_handle_ptr;
+
+  blas_benchmark::utils::CUDAScalar<scalar_t> d_d1(d1);
+  blas_benchmark::utils::CUDAScalar<scalar_t> d_d2(d2);
+  blas_benchmark::utils::CUDAScalar<scalar_t> d_x1(x1);
+  blas_benchmark::utils::CUDAScalar<scalar_t> d_y1(y1);
+  blas_benchmark::utils::CUDAVector<scalar_t> d_param(param_size, param.data());
+
+  CUBLAS_CHECK(cublasSetPointerMode(cuda_handle, CUBLAS_POINTER_MODE_DEVICE));
+
+#ifdef BLAS_VERIFY_BENCHMARK
+  // Run a first time with a verification of the results
+  scalar_t d1_ref = d1;
+  scalar_t d2_ref = d2;
+  scalar_t x1_ref = x1;
+  scalar_t y1_ref = y1;
+  std::vector<scalar_t> param_ref = param;
+
+  scalar_t d1_verify{d1};
+  scalar_t d2_verify{d2};
+  scalar_t x1_verify{x1};
+  scalar_t y1_verify{y1};
+  std::vector<scalar_t> param_verify(param);
+  reference_blas::rotmg(&d1_ref, &d2_ref, &x1_ref, &y1_ref, param_ref.data());
+
+  {
+    // Temp result on device copied back upon destruction
+    blas_benchmark::utils::CUDAScalar<scalar_t, true> d_d1_verify(d1_verify);
+    blas_benchmark::utils::CUDAScalar<scalar_t, true> d_d2_verify(d2_verify);
+    blas_benchmark::utils::CUDAScalar<scalar_t, true> d_x1_verify(x1_verify);
+    blas_benchmark::utils::CUDAScalar<scalar_t, true> d_y1_verify(y1_verify);
+    blas_benchmark::utils::CUDAVector<scalar_t, true> d_param_verify(
+        param_size, param_verify.data());
+    CUDA_CHECK(cudaDeviceSynchronize());
+    cublas_routine<scalar_t>(cuda_handle, d_d1_verify, d_d2_verify, d_x1_verify,
+                             d_y1_verify, d_param_verify);
+    CUDA_CHECK(cudaDeviceSynchronize());
+  }
+
+  std::ostringstream err_stream;
+  const bool isAlmostEqual = utils::almost_equal(d1_verify, d1_ref) &&
+                             utils::almost_equal(d2_verify, d2_ref) &&
+                             utils::almost_equal(x1_verify, x1_ref) &&
+                             utils::almost_equal(y1_verify, y1_ref) &&
+                             utils::compare_vectors(param_verify, param_ref);
+
+  if (!isAlmostEqual) {
+    err_stream << "Value mismatch." << std::endl;
+    const std::string& err_str = err_stream.str();
+    state.SkipWithError(err_str.c_str());
+    *success = false;
+  };
+#endif
+
+  auto blas_warmup = [&]() -> void {
+    cublas_routine<scalar_t>(cuda_handle, d_d1, d_d2, d_x1, d_y1, d_param);
+    CUDA_CHECK(cudaDeviceSynchronize());
```

Comment:
Replace this with `cudaStreamSynchronize` and see if it works or not. If yes, remove all device sync with stream sync.